js/apps/admin-ui/src/identity-providers/add/ExtendedOAuth2Settings.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  HelpItem,
+  KeycloakSelect,
+  SelectVariant,
+} from "@keycloak/keycloak-ui-shared";
+import {
+  ExpandableSection,
+  Form,
+  FormGroup,
+  NumberInput,
+  SelectOption,
+} from "@patternfly/react-core";
+import { useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { FormGroupField } from "../component/FormGroupField";
+import { SwitchField } from "../component/SwitchField";
+import { TextField } from "../component/TextField";
+
+const promptOptions = {
+  unspecified: "",
+  none: "none",
+  consent: "consent",
+  login: "login",
+  select_account: "select_account",
+};
+
+export const ExtendedOAuth2Settings = () => {
+  const { t } = useTranslation();
+  const { control } = useFormContext();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [promptOpen, setPromptOpen] = useState(false);
+
+  return (
+    <ExpandableSection
+      toggleText={t("advanced")}
+      onToggle={() => setIsExpanded(!isExpanded)}
+      isExpanded={isExpanded}
+    >
+      <Form isHorizontal>
+        <TextField field="config.defaultScope" label="scopes" />
+        <FormGroupField label="prompt">
+          <Controller
+            name="config.prompt"
+            defaultValue=""
+            control={control}
+            render={({ field }) => (
+              <KeycloakSelect
+                toggleId="prompt"
+                onToggle={() => setPromptOpen(!promptOpen)}
+                onSelect={(value) => {
+                  field.onChange(value as string);
+                  setPromptOpen(false);
+                }}
+                selections={field.value || t(`prompts.unspecified`)}
+                variant={SelectVariant.single}
+                aria-label={t("prompt")}
+                isOpen={promptOpen}
+              >
+                {Object.entries(promptOptions).map(([key, val]) => (
+                  <SelectOption
+                    selected={val === field.value}
+                    key={key}
+                    value={val}
+                  >
+                    {t(`prompts.${key}`)}
+                  </SelectOption>
+                ))}
+              </KeycloakSelect>
+            )}
+          />
+        </FormGroupField>
```

Comment:
Here you can use a `SelectControl`
```suggestion
     <SelectControl
            name="config.prompt"
            label={t("prompt")}
            options={[
              { key: t("prompts.unspecified"), value: "" },
              { key: t("prompts.none"), value: "none" },
              { key: t("prompts.consent"), value: "consent" },
              { key: t("prompts.login"), value: "login" },
              { key: t("prompts.select_account"), value: "select_account" },
            ]}
            controller={{ defaultValue: "" }}
          />
```